test_fms/mpp/test_mpp_init_logfile.F90

```diff
@@ -0,0 +1,44 @@
+!***********************************************************************
+!*                   GNU Lesser General Public License
+!*
+!* This file is part of the GFDL Flexible Modeling System (FMS).
+!*
+!* FMS is free software: you can redistribute it and/or modify it under
+!* the terms of the GNU Lesser General Public License as published by
+!* the Free Software Foundation, either version 3 of the License, or (at
+!* your option) any later version.
+!*
+!* FMS is distributed in the hope that it will be useful, but WITHOUT
+!* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+!* FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+!* for more details.
+!*
+!* You should have received a copy of the GNU Lesser General Public
+!* License along with FMS.  If not, see <http://www.gnu.org/licenses/>.
+!***********************************************************************
+
+!> @file
+!! @brief Tests the mpp_init_logfile interface
+!! @author Miguel Zuniga
+!! @email gfdl.climate.model.info@noaa.gov
+
+program test_mpp_init_logfile
+#include <fms_platform.h>
+
+  use mpp_mod, only : mpp_init, mpp_init_logfile
+  use mpp_mod, only : mpp_init_test_logfile_init
+
+  IMPLICIT NONE
+
+  integer :: my_mpp_init_level
+  integer :: err_no
+
+  !Initialize mpp but have mpp_init() exit before log files are initalized.
+  my_mpp_init_level = mpp_init_test_logfile_init - 1
```

Comment:
Why are you subtracting 1?  Isn't there a `mpp_init_test` level that you can directly use?

Reply:
I was subtracting one so that initialization stops before calling init_logfile. But given the 
challenge of keeping init_logfile private, and upon inspection of the mpp_init routine, it occurs to me that I can solve both issues just by not subtracting one and not calling init_log_file.
Thanks.

Reply:
I am not sure and am requesting input. If the init level is set to "mpp_init_test_logfile_init".
then mpp_init() actually does more work AFTER calling mpp_initlogfile: it writes to the logifle. So technically, to test the interface, one should isolate the interface to whatever extent possible, not after what it creates is used by another function.

Reply:
Its changed so it not subtracting now. Corresponding changed in the shell script were made. Mpp_init_logfile was removed from public (i.e. put back to original form).